tests/system/feature_store/test_feature_store.py

```diff
@@ -1923,7 +1924,42 @@ def test_purge(self):
 
         fs.ingest(fset, source)
 
-        targets_to_purge = targets[:-1]
+        targets_to_purge = (
+            targets if NoSqlTarget() not in targets else targets.remove(NoSqlTarget())
```

Comment:
This will not work - `NoSqlTarget()` generates a new instance, and it's not going to be in `targets`. So, it will return all targets, which is not the expected behavior.

Reply:
fixed.
I think i'm not clear on the test logic.
1. why do we ingest twice - what is checked in the second that that verify_purge() does not check in the first?
2. specifically in verify_purge - looking at debug run, seems the second iteration loop on fset.status.targets is skipped (on both v3io and redis) - is this expected?

Reply:
I'm not sure exactly what led to the `test_purge` as it's written. My best effort at answers is:
1. Looks like the test ingests, then purges, then ingests again after the purge, and finally purges just a subset of the targets. My best guess is that it first validates that ingest after a purge doesn't cause any issues (that's why it ingests twice)
2. I think this is in purpose to see if purge on a subset of the targets really works. But, not sure...